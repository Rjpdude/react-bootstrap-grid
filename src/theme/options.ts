export interface ThemeOptions {
  /**
   * The `margin` and `padding` in pixels for all grid elements.
   *
   * @default 15
   */
  spacing: number
  /**
   * The maximum width in pixels of the small breakpoint container.
   *
   * @default 576
   */
  sm: number
  /**
   * The width in pixels to assign to the small breakpoint container.
   *
   * @default 540
   */
  smWidth: number
  /**
   * The maximum width in pixels of the medium breakpoint container.
   *
   * @default 768
   */
  md: number
  /**
   * The width in pixels to assign to the medium breakpoint container.
   *
   * @default 720
   */
  mdWidth: number
  /**
   * The maximum width in pixels of the large breakpoint container.
   *
   * @default 992
   */
  lg: number
  /**
   * The width in pixels to assign to the large breakpoint container.
   *
   * @default 960
   */
  lgWidth: number
  /**
   * The maximum width in pixels of the extra-large breakpoint container.
   *
   * @default 1200
   */
  xl: number
  /**
   * The width in pixels to assign to the extra-large breakpoint container.
   *
   * @default 1140
   */
  xlWidth: number
}

export function getDefaultTheme(): ThemeOptions {
  return {
    spacing: 15,
    sm: 576,
    smWidth: 540,
    md: 768,
    mdWidth: 720,
    lg: 992,
    lgWidth: 960,
    xl: 1200,
    xlWidth: 1140
  }
}
