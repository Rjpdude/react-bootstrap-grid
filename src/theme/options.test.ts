import * as Options from './options'

describe('Theme Options', () => {
  test('Default theme function', () => {
    expect(Options.getDefaultTheme()).toMatchSnapshot()
  })
})
