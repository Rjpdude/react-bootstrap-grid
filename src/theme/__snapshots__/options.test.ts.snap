// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Options Default theme function 1`] = `
Object {
  "lg": 992,
  "lgWidth": 960,
  "md": 768,
  "mdWidth": 720,
  "sm": 576,
  "smWidth": 540,
  "spacing": 15,
  "themeStyle": true,
  "xl": 1200,
  "xlWidth": 1140,
}
`;
